pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPU.hpp

```diff
@@ -195,11 +195,11 @@ class AdjointJacobian final
 
         /// Ensure the new local jacs are inserted and
         /// overwrite the 0 jacs values before returning
-        for (std::size_t i = 0; i < futures.size(); i++) {
+        for (std::size_t i = 0; i < jac_futures.size(); i++) {
             const auto first = static_cast<std::size_t>(
                 std::ceil(obs.size() * i / num_chunks));
 
-            auto jac_chunk = futures[i].get();
+            auto jac_chunk = jac_futures[i].get();
             for (std::size_t j = 0; j < jac_chunk.size(); j++) {
                 std::copy(jac_chunk.begin(), jac_chunk.end(),
```

Comment:
As we would like to use the incoming  jac span? Can we add a //TODO here for it?